EpsilonAuxBMS/Src/SetContactorsTask.c

```diff
@@ -319,47 +330,115 @@ int isContactorSet(uint16_t pin, GPIO_TypeDef* port, int current_multiplier)
 /*
   This function takes multiple reads because the current amplifier has a bit of a setup time
   to make sure that erroneous values aren't read, multiple reads are taken till we have a
-  relatively steady value. Dan suggested that 3 good reads in a row is good.
+  relatively steady value.
 */
 uint32_t readCurrentThroughContactors(void)
 {
     // Enable current amplifier. Allow time to settle and take multiple valid concurrent readings
     // to make sure it isn't changing
     HAL_GPIO_WritePin(CURRENT_SENSE_ENABLE_GPIO_Port, CURRENT_SENSE_ENABLE_Pin, GPIO_PIN_SET);
-    uint32_t sense = 0;
-    uint32_t prev_sense = 0;
+    uint32_t senses[NUM_SENSES];
+    int count = 0;
 
-    for (int counter = 0; counter < 3; counter++)
+    // Want to limit it so we don't get an infinite loop
+    while (count < MAX_CURRENT_READ_ATTEMPTS)
     {
+        resetSenseArray(senses);
         osDelay(SENSE_SETUP_TIME);
+        int ret_val = isSenseStable(senses);
```

Comment:
Does ret_value stand for return value? maybe worth to rename to just `senseStable`